sonar-php-plugin/src/main/java/org/sonar/plugins/php/phpunit/CompatibilityImportersFactory.java

```diff
@@ -0,0 +1,126 @@
+import org.sonar.api.measures.CoreMetrics;
+import org.sonar.api.utils.Version;
+import org.sonar.plugins.php.PhpPlugin;
+
+public class CompatibilityImportersFactory {
+
+  public static final String DEPRECATION_WARNING_TEMPLATE = "%s is deprecated as of SonarQube 6.2. Please consider using " + PhpPlugin.PHPUNIT_COVERAGE_REPORT_PATHS_KEY;
+  public static final String SKIPPED_WARNING_TEMPLATE = "Ignoring %s since you are already using " + PhpPlugin.PHPUNIT_COVERAGE_REPORT_PATHS_KEY + ". Please remove %<s";
+  public static final String NOT_YET_SUPPORTED_WARNING = PhpPlugin.PHPUNIT_COVERAGE_REPORT_PATHS_KEY + " is only supported by SonarQube 6.2 and above";
+  private static final List<String> LEGACY_PATH_KEYS = Arrays.asList(PhpPlugin.PHPUNIT_COVERAGE_REPORT_PATH_KEY, PhpPlugin.PHPUNIT_IT_COVERAGE_REPORT_PATH_KEY,
+    PhpPlugin.PHPUNIT_OVERALL_COVERAGE_REPORT_PATH_KEY);
+  private final SensorContext context;
+
+  public CompatibilityImportersFactory(SensorContext context) {
+    this.context = context;
+  }
+
+  public List<ReportImporter> createCoverageImporters() {
+    if (!supportsMultiPathCoverage()) {
+      return createLegacyImporters();
+    }
+    if (LEGACY_PATH_KEYS.stream().anyMatch(this::isPropertyUsed) && !multiPathCoverageUsed()) {
+      return createLegacyImporters();
+    }
+    return createMultiCoverageImporter();
+  }
+
+  private boolean supportsMultiPathCoverage() {
+    return context.getSonarQubeVersion().isGreaterThanOrEqual(Version.create(6, 2));
+  }
+
+  private boolean multiPathCoverageUsed() {
+    return context.settings().getStringArray(PhpPlugin.PHPUNIT_COVERAGE_REPORT_PATHS_KEY).length > 0;
+  }
+
+  private static List<ReportImporter> createMultiCoverageImporter() {
+    String msg = "coverage";
+    String propertyKey = PhpPlugin.PHPUNIT_COVERAGE_REPORT_PATHS_KEY;
+
+    CoverageResultImporter singleReportImporter = new CoverageResultImporter(propertyKey,
+      msg,
+      CoreMetrics.LINES_TO_COVER,
+      CoreMetrics.UNCOVERED_LINES,
+      CoverageType.UNIT);
+
+    return Collections.singletonList(new MultiPathImporter(singleReportImporter, propertyKey, msg));
+  }
+
+  private static List<ReportImporter> createLegacyImporters() {
+    List<ReportImporter> importers = new ArrayList<>();
+
+    importers.add(new CoverageResultImporter(
+      PhpPlugin.PHPUNIT_COVERAGE_REPORT_PATH_KEY,
+      "unit test coverage",
+      CoreMetrics.LINES_TO_COVER,
+      CoreMetrics.UNCOVERED_LINES,
+      CoverageType.UNIT));
+
+    importers.add(new CoverageResultImporter(
+      PhpPlugin.PHPUNIT_IT_COVERAGE_REPORT_PATH_KEY,
+      "integration test coverage",
+      CoreMetrics.IT_LINES_TO_COVER,
+      CoreMetrics.IT_UNCOVERED_LINES,
+      CoverageType.IT));
+
+    importers.add(new CoverageResultImporter(
+      PhpPlugin.PHPUNIT_OVERALL_COVERAGE_REPORT_PATH_KEY,
+      "overall coverage",
+      CoreMetrics.OVERALL_LINES_TO_COVER,
+      CoreMetrics.OVERALL_UNCOVERED_LINES,
+      CoverageType.OVERALL));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "CoverageType"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "OVERALL"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)